pkg/clients/aws/route53.go

```diff
@@ -123,13 +126,23 @@ func (c *awsClient) AnswerDNSChallenge(reqLogger logr.Logger, acmeChallengeToken
 		return "", err
 	}
 
-	baseDomain := cr.Spec.ACMEDNSDomain
-	if !strings.HasSuffix(baseDomain, ".") {
-		baseDomain = baseDomain + "."
+	dnsZones := hivev1.DNSZoneList{}
+	err = c.kubeClient.List(context.TODO(), &dnsZones, &client.ListOptions{Namespace: c.namespace})
+	if err != nil {
+		reqLogger.Error(err, err.Error())
+		return "", err
+	}
+
+	if len(dnsZones.Items) == 0 {
+		return "", fmt.Errorf("no dns zones found")
 	}
+	dnsZone := dnsZones.Items[0]
+
+	dnsZonePath := dnsZone.Status.AWS.ZoneID
+	dnsZoneId := filepath.Base(*dnsZonePath)
 
 	for _, hostedzone := range hostedZones {
-		if strings.EqualFold(baseDomain, *hostedzone.Name) {
+		if strings.EqualFold(dnsZoneId, *hostedzone.Id) {
 			zone, err := c.client.GetHostedZone(&route53.GetHostedZoneInput{Id: hostedzone.Id})
```

Comment:
Do we even need this loop/if anymore? We know exactly what the ID should be and can just `route53:GetHostedZone` with `dnszone.Status.AWS.ZoneID`?